atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -295,8 +313,20 @@ public void setUp() throws Exception {
     }
 
     @Override
-    protected TestTransactionManager wrapTestTransactionManager(TestTransactionManager testTransactionManager) {
-        return new WrappingTestTransactionManagerImpl(testTransactionManager, transactionWrapper);
+    protected TestTransactionManager constructTestTransactionManager() {
+        return new WrappingTestTransactionManagerImpl(
```

Comment:
why do we need to have a `constructTetsTransactionManager`, is it not sufficient to wrap as before? somewhat confused

Reply:
Wrapping doesn't work since it when we switch to delegate we lose control how transactions are created. Refactored `TestTransactionManagerImpl` to solve this.